### DISCUSSION TITLE
[UX] Allow to navigate from direct link to homepage

### DISCUSSION THREAD

**Discussion by KaKi87** [Docile-Alligator/Infinity-For-Reddit#921] [Ideas]
Hello,

When clicking on a direct Reddit link (e.g. : Google search result, social media post, etc.), it is possible to navigate to the subreddit but not to the homepage.

Therefore I would like to suggest repurposing the top-left "back" button for going "up", aka from post to subreddit then from subreddit to homepage.

Meanwhile, Android's native "back" navigation would continue to work as always.

Thanks

**Comment by Docile-Alligator** [comment]
Nice suggestion!

**Comment by KaKi87** [comment]
Hello, any news on this ? Thanks !

**Comment by Wladefant** [comment]
Hey @KaKi87 so you mean if i open a post from my feed and then tap on the back button it should open the subreddit and then the homepage, and if i tap on the android back button it should perform the last action (last post...)?

**Reply by KaKi87** [reply]
> i open a post from my feed and then tap on the back button it should open the subreddit and then the homepage

Yes, more specifically : feed -> post -> [UP] -> subreddit -> [UP] -> feed
> i tap on the android back button it should perform the last action (last post...)

No : feed -> [BACK] -> subreddit -> [BACK] -> post -> [BACK] -> feed

(i.e. the opposite of the previous part)

If you're curious, the [OctoDroid](https://github.com/slapperwan/gh4a) GitHub client app works exactly like I'm describing.
